sources/dogtag/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+/*!
+dogtag is a set of tools that detect the hostname of a bottlerocket server/instance and prints it to stdout.
+if the tool is called in an environment it cannot resolve the hostname it will error out.
+
+Currently the following hostname tools are implemented:
+
+* 01-imds - Fetches hostname from the Instance Metadata via IMDS
+* 00-reverse-dns - Uses reverse dns lookup to resolve the hostname
+ */
+use argh::FromArgs;
+use log::debug;
+use snafu::ResultExt;
+use std::{fs, process};
+
+const DOGTAG_BIN_PATH: &str = "/var/bottlerocket/dogtag";
+
+/// Cli defines the standard cmdline interface for all hostname handlers
+#[derive(FromArgs)]
+#[argh(description = "hostname resolution tool")]
+pub struct Cli {
+    #[argh(option)]
+    #[argh(description = "ip_address of the host")]
+    pub ip_address: String,
+}
+
+pub type Result<T> = std::result::Result<T, error::Error>;
+
+/// find_hostname will utilize the helpers located in /var/bottlerocket/dogtag/ to try and discover the hostname
+pub async fn find_hostname() -> Result<String> {
+    debug!(
+        "attempting to discover hostname helpers in {}",
+        DOGTAG_BIN_PATH
+    );
+    let walker = fs::read_dir(DOGTAG_BIN_PATH).context(error::IoSnafu)?;
```

Comment:
you may be able to use `walkdir` in order to gather and sort the binaries all in one step, e.g. https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/early-boot-config/early-boot-config/src/main.rs#L96-L106